### ISSUE TITLE
pysot分布式训练的问题

### ISSUE THREAD

**Issue by Jontyshaw** [STVIR/pysot#598]
def reduce_gradients(model, _type='sum'):
    types = ['sum', 'avg']
    assert _type in types, 'gradients method must be in "{}"'.format(types)
    log_once("gradients method is {}".format(_type))
    if get_world_size() > 1:
        for param in model.parameters():
            if param.requires_grad:
                dist.all_reduce(param.grad.data)
                if _type == 'avg':
                    param.grad.data /= get_world_size()
    else:
        return None
在分布式训练环境dist.all_reduce(param.grad.data)这部分代码报错，报错为AttributeError: 'NoneType' object has no attribute 'data'

**Comment by yarkable** [comment]
解决了吗

**Comment by sssimpleboy** [comment]
> 解决了吗

解决了吗？  我前几周还没问题  今天出现了 哈哈哈

**Comment by yarkable** [comment]
> > 解决了吗
> 
> 解决了吗？ 我前几周还没问题 今天出现了 哈哈哈

解决了，有些没有required_grad的tensor就加个if把它过滤掉，不要走到这个分支

**Comment by sssimpleboy** [comment]
> > > 解决了吗
> > 
> > 
> > 解决了吗？ 我前几周还没问题 今天出现了 哈哈哈
> 
> 解决了，有些没有required_grad的tensor就加个if把它过滤掉，不要走到这个分支

嗯嗯，我是把模型改了，结果配置文件忘记改了
